packages/mui-material/src/Modal/ModalManager.ts

```diff
@@ -56,22 +56,56 @@ function isAriaHiddenForbiddenOnElement(element: Element): boolean {
   return isForbiddenTagName || isInputHidden;
 }
 
-function ariaHiddenSiblings(
+function ariaHiddenElements(
   container: Element,
-  mountElement: Element,
+  mountElement: Element | null,
   currentElement: Element,
   elementsToExclude: readonly Element[],
   show: boolean,
 ): void {
-  const blacklist = [mountElement, currentElement, ...elementsToExclude];
+  let current: Element | null = container;
+  let previousElement: Element =
+    container === mountElement ? currentElement : (mountElement ?? currentElement);
+  const html = ownerDocument(container).body.parentElement;
+  const blacklist = [mountElement, ...elementsToExclude];
+
+  // In theory this should not happen anymore.
+  // in some cases the container and previous element still
+  // could end up being the same, in this case we just go up 1
+  if (current === previousElement) {
+    current = current.parentElement;
+  }
```

Comment:
When could this edge case happen? Can you give a real-world example?

Reply:
It's been like 2 years since I wrote this, it can happen if the container and the current element passed in are the same, but I don't remember where it was called like that.

I think I fixed it but I kept the check in just in case so this function always exits properly no matter how it's called (it's a while loop after all).

Reply:
That’s defensive. If the container and the starting element are the same, it means the container resolution logic is flawed. When we trust that the DOM structure is correct, so we shouldn’t include this. I am removing it.

Reply:
It was misused in the library itself before so yes, it is defensive because I can't guarantee that someone will not make the same mistake again in the future.

It's too late to modify the runtime behavior to throw/warn to defend against when someone does this - I don't want to break apps depending on material-ui after all.